web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -739,9 +732,11 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
     let kbdAspectRatio = layerGroup.offsetWidth / this.kbdDiv.offsetHeight;
     let baseKeyProbabilities = this.kbdLayout.getLayer(this.layerId).getTouchProbabilities(touchKbdPos, kbdAspectRatio);
 
-    if (!keySpec || !this.subkeyGesture || !this.subkeyGesture.baseKey.key) {
+    if (!keySpec /*|| !this.subkeyGesture || !this.subkeyGesture.baseKey.key*/) {
       return baseKeyProbabilities;
-    } else {
+    }
+    // TODO:  predictive-text will need mild rework for integration with new gesture integration patterns.
```

Comment:
One of the noted TODOs.

To remove the old `PendingGesture` and `RealizedGesture` classes, which were referencing a now-defunct type connected to the old engines, I needed to comment out this code.  It wasn't being triggered as of #9698, but that in itself is likely a bit of an oversight.